### PR TITLE
Introduce rule tables in rules page

### LINKED ISSUE #6928: Introduce rules tables in rules page
### What is the problem you're trying to solve?

We currently use the `standard` preset. Several times, I skimmed through the whole list of rules to see which extra rules need to be manually enabled. Because the Ⓡ & Ⓢ characters blend in so well on the page, it can't be easily done by skimming visually, and one misses a rule easily.

### What solution would you like to see?

I can think of 2 solutions, neither mutually exclusive.

The first idea and reason why I opened this issue, would be a couple toggles at the top of the page to show/hide rules that are already in a preset. That way the list would only contain rules that one needs to switch on manually.

In comparison, the list of ESLint rules is a lot easier to visually parse: https://eslint.org/docs/latest/rules/
So a second option could be to move align those characters on the right of the list, with maybe some color coding, spacing, etc. to make everything stand out a little.

Thanks for this amazing tool!

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Closes #6928

See https://github.com/stylelint/stylelint.io/pull/328 to preview the website.

> Is there anything in the PR that needs further explanation?

This change introduces rule tables to the rules page to improve visibility.

### CODE REVIEW

File: lib/rules/__tests__/index.test.js [hunk 1]
```diff
@@ -58,3 +59,45 @@ describe('custom message option', () => {
...
 });
+
+describe('recommended and standard configs', () => {
+	const cwd = path.join(__dirname, 'tmp');
+
+	fs.rmSync(cwd, { recursive: true, force: true });
+	fs.mkdirSync(cwd, { recursive: true });
+	fs.writeFileSync(path.join(cwd, 'package.json'), '{}');
+	cp.execSync(
+		'npm install --silent --no-package-lock --no-audit --omit=peer stylelint-config-recommended stylelint-config-standard',
```
Comment (ybiquitous) [inline lib/rules/__tests__/index.test.js:70]: [note] For details about options, see https://docs.npmjs.com/cli/v9/commands/npm-install

File: docs/user-guide/rules.md [hunk 2]
```diff
@@ -5,7 +5,11 @@ There are over one hundred built-in rules to help you:
 - [avoid errors](#avoid-errors)
 - [enforce conventions](#enforce-conventions)
 
-Many of them are turned on in our [recommended](https://www.npmjs.com/package/stylelint-config-recommended) (Ⓡ) and [standard](https://www.npmjs.com/package/stylelint-config-standard) (Ⓢ) configs.
+Many of them are turned on in our [recommended](https://www.npmjs.com/package/stylelint-config-recommended) and [standard](https://www.npmjs.com/package/stylelint-config-standard) configs.
+
+- ⭐ - [Recommended](https://www.npmjs.com/package/stylelint-config-recommended)
```
Comment (romainmenke) [inline docs/user-guide/rules.md:10]: `⭐` is rendered differently.

Safari :

Screenshot 2023-06-19 at 23 07 55

Chrome :

Screenshot 2023-06-19 at 23 08 08

Firefox : 

Screenshot 2023-06-19 at 23 08 31

----

This should work everywhere : `&#x2b50;&#xFE0F;`

copy paste this one : ⭐️

Comment (ybiquitous) [reply docs/user-guide/rules.md]: Thank you! I missed it. Fixed via d2b54e3e.

**Review by romainmenke — APPROVED** [review]
LGTM!

Thank you for the very large update to `rules.md`.
Extra tests are also very nice!

File: docs/user-guide/rules.md [hunk 3]
```diff
@@ -15,87 +19,107 @@ You can avoid errors with these `no` rules.
...
+| [`no-duplicate-selectors`](../../lib/rules/no-duplicate-selectors/README.md)<br/>Disallow duplicate selectors.                                                                                       | ✅  | ✅  |     |
 
 ### Empty
 
 Disallow empty things with these `no-empty` rules.
 
-- [`block-no-empty`](../../lib/rules/block-no-empty/README.md): Disallow empty blocks (Ⓡ & Ⓢ).
-- [`comment-no-empty`](../../lib/rules/comment-no-empty/README.md): Disallow empty comments (Ⓡ & Ⓢ).
-- [`no-empty-source`](../../lib/rules/no-empty-source/README.md): Disallow empty sources (Ⓡ & Ⓢ).
+| Rule                                                                                          | ⭐️ | 💅  | 🔧  |
```
Comment (Mouvedia) [inline docs/user-guide/rules.md:43]: too bad we can't add a tooltip (title attribute) on the icons

Comment (ybiquitous) [reply docs/user-guide/rules.md:43]: I don't think Markdown supports the `title` attribute. But we can add anything on the website. I'll try it on stylelint.io

Comment (ybiquitous) [reply docs/user-guide/rules.md:43]: See https://github.com/stylelint/stylelint.io/pull/328/commits/91ef25bda66435f08e33ad41473e6a23d47bb5e1

**Pull request merged**